### ISSUE TITLE
bug: round trip `FixedSizeList` lance <-> arrow. nullability is not preserved

### ISSUE THREAD

**Issue by universalmind303** [lance-format/lance#2304]
round tripping native lance `Field` from `ArrowField` and back drops the `nullable` field. 

```rs
let arrow_field = ArrowField::new_fixed_size_list(
    "list",
    ArrowField::new("item", DataType::Int32, false),
    1,
    false,
);
let field = Field::try_from(&arrow_field).unwrap();

assert_eq!(field.name, "list");
assert_eq!(field.nullable, arrow_field.is_nullable());
assert_eq!(&field.data_type(), arrow_field.data_type());
assert_eq!(ArrowField::from(&field), arrow_field);
```

it seems that we drop it because `Field::data_type`  calls `impl TryFrom<&LogicalType> for DataType`, which [hard codes](https://github.com/lancedb/lance/blob/7c8333f903a0d86dc2f3af0dca0d0160d7700cb3/rust/lance-core/src/datatypes.rs#L226) the `nullable` field.

**Comment by wjones127** [comment]
Yeah the underlying problem is that when we implemented FixedSizeList data type, we decided not to make the inner field an actual field. It just serialized the data type, and doesn't handle any metadata, nullability, or child fields. This also blocks #1293

I started to fix this in #1693, but it causes drastic breaking changes in how schemas and page tables are handled.

**Comment by universalmind303** [comment]
> I started to fix this in https://github.com/lancedb/lance/pull/1693, but it causes drastic breaking changes in how schemas and page tables are handled.

 is it possible to do this in a non breaking way?
